### PR TITLE
Refactor entry modules - Closes #3580

### LINKED ISSUE #3580: Move state management and entry modules to the top level
### Description

- Loader
- Delegates
- Transport
should be moved to the root level, and all the repeating task based on time should be managed in the chain.js to avoid circular dependencies.

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
Entry modules was in the same level as other modules and had circular dependency

### How did I fix it?
Move 3 entry modules to root level, delegates, transport and loader.
Also, split delegates management and forging into 2 files.

### How to test it?
- Run the tests
- start the application

### Review checklist

* The PR resolves #3580 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/chain.js [hunk 1]
```diff
@@ -152,6 +155,19 @@ module.exports = class Chain {
 			this.scope.bus = await createBus();
 			this.scope.logic = await initLogicStructure(this.scope);
 			this.scope.modules = await initModules(this.scope);
+			// TODO: Global variable forbits to require on top
+			const Loader = require('./loader');
+			const { Forge } = require('./forge');
```
Comment (nazarhussain) [inline framework/src/modules/chain/chain.js]: `forge` is a verb not correct for a module name. Rename it to `forger`

File: framework/src/modules/chain/chain.js [hunk 2]
```diff
@@ -152,6 +155,19 @@ module.exports = class Chain {
...
+			const { Delegates } = require('./submodules/delegates');
+			const Transport = require('./transport');
+			this.loader = new Loader(this.scope);
+			this.forge = new Forge(this.scope);
+			this.transport = new Transport(this.scope);
+			// TODO: should not add to scope
+			this.scope.modules.delegates = new Delegates(this.scope);
+			this.scope.modules.loader = this.loader;
+			this.scope.modules.forge = this.forge;
+			this.scope.modules.transport = this.transport;
```
Comment (nazarhussain) [inline framework/src/modules/chain/chain.js]: If these are modules then what's wrong to load and return from in `init_modules` step?

File: framework/src/modules/chain/chain.js [hunk 3]
```diff
@@ -163,7 +179,14 @@ module.exports = class Chain {
 			// Fire onBind event in every module
 			this.scope.bus.message('bind', this.scope);
 
-			self.logger.info('Modules ready and launched');
+			// After binding, it should immeately load blockchain
```
Comment (nazarhussain) [inline framework/src/modules/chain/chain.js]:
```suggestion
			// After binding, it should immediately load blockchain
```

File: framework/src/modules/chain/chain.js [hunk 4]
```diff
@@ -163,7 +179,14 @@ module.exports = class Chain {
...
+			await this.loader.loadBlockChain();
+			this.logger.info('Modules ready and launched');
```
Comment (nazarhussain) [inline framework/src/modules/chain/chain.js]: This seems the wrong order. Blockchain is ready or modules?

File: framework/src/modules/chain/loader.js [hunk 5]
```diff
@@ -17,28 +17,22 @@
 const async = require('async');
```
Comment (nazarhussain) [inline framework/src/modules/chain/loader.js]: I don't understand how moving different modules of chain to one level up make any difference. Same you can do but keeping the files where these belongs to.

File: framework/src/modules/chain/chain.js [hunk 6]
```diff
@@ -295,4 +318,64 @@ module.exports = class Chain {
 
 		this.logger.info('Cleaned up successfully');
 	}
+
+	_startLoader() {
```
Comment (nazarhussain) [inline framework/src/modules/chain/chain.js]: Move these actions to their respective modules, don't thick the initialization logic file.

File: framework/src/modules/chain/chain.js [hunk 7]
```diff
@@ -295,4 +318,64 @@ module.exports = class Chain {
...
+						if (syncError) {
+							this.logger.error('Sync timer', syncError);
+						}
+					}
+				);
+			}
+		}, syncInterval);
+	}
+
+	async _startForge() {
```
Comment (nazarhussain) [inline framework/src/modules/chain/chain.js]: Same here

File: framework/src/modules/chain/submodules/blocks/process.js [hunk 8]
```diff
@@ -347,28 +347,11 @@ class Process {
 	/**
 	 * Handle newly received block.
 	 *
-	 * @listens module:transport~event:receiveBlock
 	 * @param {block} block - New block
 	 * @todo Add @returns tag
 	 */
 	// eslint-disable-next-line class-methods-use-this
-	onReceiveBlock(block) {
```
Comment (nazarhussain) [inline framework/src/modules/chain/submodules/blocks/process.js]: Are we going to remove this `on<Event>` convention? If that's the case then there are many `onBind` handlers in the code.

File: framework/test/mocha/unit/modules/chain/transport.js [hunk 9]
```diff
@@ -23,21 +23,19 @@ const {
...
+const Block = require('../../../fixtures/blocks').Block;
 const {
 	registeredTransactions,
-} = require('../../../../common/registered_transactions');
-const InitTransaction = require('../../../../../../src/modules/chain/logic/init_transaction');
-const ProcessTransactions = require('../../../../../../src/modules/chain/submodules/process_transactions');
-const processTransactionLogic = require('../../../../../../src/modules/chain/logic/process_transaction');
+} = require('../../../common/registered_transactions');
+const InitTransaction = require('../../../../../src/modules/chain/logic/init_transaction');
+const ProcessTransactions = require('../../../../../src/modules/chain/submodules/process_transactions');
```
Comment (nazarhussain) [inline framework/test/mocha/unit/modules/chain/transport.js]: `ProcessTransactions` is really not a good name for a module or a class. It should be renamed to `TransactionsProcessor` or `TransactionsLoader`

File: framework/src/modules/chain/chain.js (hunk 2 above)
Comment (shuse2) [reply framework/src/modules/chain/chain.js]: It's still temporally, and I want to remove this `init_modules` step afterwards

File: framework/src/modules/chain/loader.js (hunk 5 above)
Comment (shuse2) [reply framework/src/modules/chain/loader.js]: For more clarity of dependency tree, but a lot more coming afterwards

File: framework/src/modules/chain/chain.js (hunk 6 above)
Comment (shuse2) [reply framework/src/modules/chain/chain.js]: It's not initialisation, chain.js will be orchestrating the modules

File: framework/src/modules/chain/submodules/blocks/process.js (hunk 8 above)
Comment (shuse2) [reply framework/src/modules/chain/submodules/blocks/process.js]: yes we are going to remove `bus` in general, but step by step

File: framework/test/mocha/unit/modules/chain/transport.js (hunk 9 above)
Comment (shuse2) [reply framework/test/mocha/unit/modules/chain/transport.js]: This will be change in coming PRs

File: framework/src/modules/chain/chain.js [hunk 10]
```diff
@@ -152,6 +155,19 @@ module.exports = class Chain {
 			this.scope.bus = await createBus();
 			this.scope.logic = await initLogicStructure(this.scope);
 			this.scope.modules = await initModules(this.scope);
+			// TODO: Global variable forbits to require on top
+			const Loader = require('./loader');
```
Comment (pablitovicente) [inline framework/src/modules/chain/chain.js]: - I think `requires` would be better at the top level of the file
- giving that this method gets too extensive maybe create a method `loadDependencies()` that does the same but then returns the value and here it's only assigned to this?

File: framework/src/modules/chain/chain.js (hunk 7 above)
Comment (pablitovicente) [inline framework/src/modules/chain/chain.js]: `_startForging` maybe ?

File: framework/src/modules/chain/chain.js (hunk 10 above)
Comment (shuse2) [reply framework/src/modules/chain/chain.js]: Let's do this loadDependencies at the end of the technical debt because it will change a lot

**Pull request merged**